Update to latest OS packages and add support for Ubuntu, Fedora, and AL 2023

jsf9k opened the pull request:
## 🗣 Description ##

This pull request updates the Ansible role to:
- Use the latest versions of the OS packages
- Support Ubuntu, Fedora, and Amazon Linux 2023

## 💭 Motivation and context ##

- We may as well use the latest versions of the OS packages as opposed to relying on CDM to update them for us.  (Also our packages were woefully outdated - two major versions behind!)
- Supporting Fedora will help us to add CrowdStrike support to [our FreeIPA server AMI](https://github.com/cisagov/freeipa-server-packer).

## 🧪 Testing ##

All automated testing passes.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

jsf9k committed:
Update the version of the OS package

jsf9k committed:
Add in Ubuntu platforms

jsf9k committed:
Add Fedora platforms

Note that we use a RedHat ES 8 OS package as we lack a
Fedora-specific package.

jsf9k committed:
Update bucket objects to which the test user is granted access

New OS packages have been issued, and we will require access to them.

dav3r commented on tasks/RedHat.yml:
```diff
@@ -0,0 +1,36 @@
+---
+- name: Grab CrowdStrike Falcon system package GPG key from S3
+  ansible.builtin.aws_s3:
+    bucket: "{{ crowdstrike_third_party_bucket_name }}"
+    object: "{{ package_gpg_key_object_name }}"
+    dest: /tmp/{{ package_gpg_key_object_name }}
+    mode: get
```
Alphabetize keys:
```suggestion
    bucket: "{{ crowdstrike_third_party_bucket_name }}"
    dest: /tmp/{{ package_gpg_key_object_name }}
    mode: get
    object: "{{ package_gpg_key_object_name }}"
```

dav3r reviewed (approved):
Looks great - only one small quibble. 👍

jsf9k committed:
Alphabetize YAML keys

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k committed:
Add support for Amazon 2023

jsf9k committed:
Prefer the newer dnf over yum

jsf9k committed:
Force install without checking GPG signature on Fedora 38 and AL 2023

The GPG signing key we have for the RPM cannot be imported on Amazon
2023.  I get the following error output:
  gpg: directory '/root/.gnupg' created
  gpg: keybox '/root/.gnupg/pubring.kbx' created
  gpg: WARNING: no command supplied.  Trying to guess what you mean ...
  gpg: no valid OpenPGP data found.
  gpg: processing message failed: Unknown system error

The GPG signing key we have for the RPM cannot be imported on Fedora
38.  I get the following error:
  error: Certificate 4727BBD5519B177F:
    The certificate is expired: The primary key is not live
  error: /tmp/Falcon Linux Sensor RPM signing GPG key: key 1 import failed.

This is why we do not attempt to import the key on these platforms and
force dnf to install the corresponding package without checking the
GPG key.

dav3r commented on vars/Amazon.yml:
```diff
@@ -0,0 +1,8 @@
+# The name of the S3 object corresponding to the CrowdStrike Falcon
+# sensor system package
+package_object_name: falcon-sensor-6.56.0-15309.amzn2023.x86_64.rpm
+
+# The name of the S3 object corresponding to the GPG key for the
+# CrowdStrike Falcon system package
+package_gpg_key_object_name: "Falcon Linux Sensor code signing certificate 2021.pem"
```
Can this be deleted since we are not using it?
```suggestion
```

dav3r replied:
Alternatively, we could comment it out and add a TODO linked to an issue.

dav3r reviewed (commented):
I have one note before I approve this.

jsf9k committed:
Remove unused variable

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

dav3r reviewed (approved):
🚀
